src/reducers/__tests__/existingThoughtMove.ts

```diff
@@ -560,3 +567,62 @@ it('move with nested duplicate thoughts and merge their children', () => {
   // context ['p', 'a', 'b'] should not have any garbage children
   expect(getChildrenRanked(stateNew, ['p', 'a', 'b'])).toHaveLength(0)
 })
+
+it('pending thoughts should be merged correctly(fetch pending before move)', async () => {
+  initialize()
+
+  const text = `
+  - a
+    - b
+      -c
+        - 1
+        - 2
+  - d
+    - b
+      - c
+        - 3
+        - 4`
+
+  timer.useFakeTimer()
+
+  appStore.dispatch([
+    importTextAction({
+      path: HOME_PATH,
+      text
+    }),
+  ]
+  )
+  await timer.runAllAsync()
+
+  timer.useFakeTimer()
+  // clear and call initialize again to reload from local db (simulating page refresh)
+  appStore.dispatch(clear())
+  await timer.runAllAsync()
+
+  initialize()
+
+  await timer.runAllAsync()
+
+  timer.useFakeTimer()
+
+  appStore.dispatch([setCursorFirstMatchActionCreator(['a'])])
+  await timer.runAllAsync()
+
+  appStore.dispatch([
+    setCursorFirstMatchActionCreator(['a']),
```

Comment:
Looks like `setCursorFirstMatchActionCreator(['a'])` is dispatched twice. Is that intentional?